src/utils/vm_image_vault_utils.cpp

```diff
@@ -32,6 +33,21 @@ QString mp::vault::filename_for(const mp::Path& path)
     return file_info.fileName();
 }
 
+QString mp::vault::copy(const QString& file_name, const QDir& output_dir)
+{
+    if (file_name.isEmpty())
+        return {};
```

Comment:
Shouldn't this be an error? What's the use case for calling copy with an empty source? :man_shrugging:

Reply:
This is actually a wholesale copy of https://github.com/canonical/multipass/blob/main/src/daemon/default_vm_image_vault.cpp#L172-L185 and I'll have a follow up PR to remove it from that file.

That said, yeah, I suppose it should throw here, but perhaps taking care of all of that should be in the follow up PR.